orchestration/templates/process-table.yaml

```diff
@@ -98,3 +98,102 @@ spec:
+                  value: '{{ "{{inputs.parameters.staging-bq-project}}" }}'
+                - name: bq-dataset
+                  value: '{{ "{{inputs.parameters.staging-bq-dataset}}" }}'
+                - name: join-table-name
+                  value: '{{ "{{tasks.diff-table.outputs.parameters.join-table-name}}" }}'
+                - name: table-name
+                  value: '{{ "{{inputs.parameters.table-name}}" }}'
+                - name: query-type
+                  value: new_rows
+          # updated rows
+          #- name: generate-updated-rows-table
+          #  dependencies: [diff-table]
+          #  template: query-full-rows
+          # arguments:
+          #   parameters:
+          #     - name: bq-project
+          #       value: '{{ "{{inputs.parameters.staging-bq-project}}" }}'
+          #     - name: bq-dataset
+          #       value: '{{ "{{inputs.parameters.staging-bq-dataset}}" }}'
+          #     - name: join-table-name
+          #       value: '{{ "{{tasks.diff-table.outputs.parameters.join-table-name}}" }}'
+          #     - name: table-name
+          #       value: '{{ "{{inputs.parameters.table-name}}" }}'
+          #     - name: query-type
+          #       value: updates
+          # deleted rows - TODO
+          #- name: generate-deleted-rows-table
+          #  dependencies: [diff-table]
+          #  template: query-row-ids
+          # arguments:
+          #   parameters:
+          #     - name: bq-project
+          #       value: '{{ "{{inputs.parameters.staging-bq-project}}" }}'
+          #     - name: bq-dataset
+          #       value: '{{ "{{inputs.parameters.staging-bq-dataset}}" }}'
+          #     - name: join-table-name
+          #       value: '{{ "{{tasks.diff-table.outputs.parameters.join-table-name}}" }}'
+          #     - name: table-name
+          #       value: '{{ "{{inputs.parameters.table-name}}" }}'
+          #     - name: query-type
+          #       value: updates
+    # Subtemplates
+    # query-full-rows will be generic for both the new rows + updates
+    - name: query-full-rows
+      inputs:
+        parameters:
+          - name: table-name
+          - name: join-table-name
+          - name: bq-project
+          - name: bq-dataset
+          - name: query-type
+      {{- include "argo.retry" . | indent 6 }}
+      script:
+        # adding a layer of environment variables to make the scripts more stand-alone, and editor compatible
+        env:
+          - name: PROJECT
+            value: '{{ "{{inputs.parameters.bq-project}}" }}'
+          - name: DATASET
+            value: '{{ "{{inputs.parameters.bq-dataset}}" }}'
+          - name: INPUT_TABLE
+            value: '{{ "{{inputs.parameters.join-table-name}}" }}'
+          - name: TABLE
+            value: '{{ "{{inputs.parameters.table-name}}" }}'
+          - name: QUERY_TYPE
+            value: '{{ "{{inputs.parameters.query-type}}" }}'
+        command: [bash]
```

Comment:
Oops, this is missing an `image` to tell it which container to use.

The base image should be `.Values.argoTemplates.diffBQTable.schemaImageName`. The version should be decided dynamically [like this](https://github.com/DataBiosphere/clinvar-ingest/blob/master/orchestration/templates/run-dataflow.yaml#L1). The final YAML line will look something [like this](https://github.com/DataBiosphere/clinvar-ingest/blob/master/orchestration/templates/run-dataflow.yaml#L16).